packages/block-editor/src/hooks/border.js

```diff
@@ -200,6 +204,56 @@ export function hasBorderSupport( blockName, feature = 'any' ) {
 	return !! support?.[ feature ];
 }
 
+/**
+ * Determine whether there is block support for shadow properties.
+ *
+ * @param {string} blockName Block name.
+ *
+ * @return {boolean} Whether there is support.
+ */
+export function hasShadowSupport( blockName ) {
+	return hasBlockSupport( blockName, SHADOW_SUPPORT_KEY );
+}
+
+/**
+ * Computes the label for the border panel based on the block's support for border and shadow,
+ * considering the blockName or defaultControls.
+ *
+ * @param {Object}             params
+ * @param {string | undefined} params.blockName The block name.
+ * @param {Object}             params.controls  The block's controls.
+ *
+ * @return {string | undefined} The label for the border panel.
+ */
+export function getBorderPanelLabel( params ) {
+	if ( ! params ) {
+		return;
+	}
+
+	const { blockName, controls } = params;
+	let supportsBorder, supportsShadow;
+
+	if ( blockName ) {
+		supportsBorder = hasBorderSupport( blockName );
+		supportsShadow = hasShadowSupport( blockName );
+	} else {
+		supportsBorder = controls?.radius || controls?.color || controls?.width;
+		supportsShadow = controls?.shadow;
```

Comment:
From global styles menu call, name prop isn't available. And the fallback should not be based on the default controls because hiding and showing might change the label.

Reply:
Fixed here b160ff0e38ef25db5de834b16a9f44d71b5791ac